src/Symfony/Component/HttpKernel/DependencyInjection/ControllerArgumentValueResolverPass.php

```diff
@@ -51,6 +67,7 @@ public function process(ContainerBuilder $container)
         $container
             ->getDefinition('argument_resolver')
             ->replaceArgument(1, new IteratorArgument($resolvers))
+            ->replaceArgument(2, new ServiceLocatorArgument($namedResolvers))
```

Comment:
this has to be setArgument :D (and replaceArgument for the previous line)

Reply:
I don’t understand, aren’t both abstract arguments? Why should they be set differently?

Reply:
because HttpKernel is decoupled from FrameworkBundle, se we cannot assume that the argument is always declared on the service side

Reply:
I see, thanks